### PR TITLE
Orifice model for cavities, ACH output, PerfectGas twinhouse and bugfixes

### PR DESCRIPTION
**Pull request by kldjonge**
Several alterations were mistakenly left on on older branch. I cherry-picked, corrected and recommitted them here.

The alterations concern:

- The use of an orifice model for the 1-port implementation of the cavity in the internal wall
- Added ACH output to each zone
- Changed shielding conditions of Twinhouse O5 to more realistic values (will only influence the pressure driven airflow simulations)
- TwinHouse05 case with Air.PerfectGas as the medium to have a better comparison with specialized airflow simulation software CONTAM. (Also showcases the use of this medium for cases were naturally driven airflows are important)
- Bugfix in the Twinhouse heating schedule that loaded the wrong initialization temperature.

### CODE REVIEW

File: IDEAS/Examples/TwinHouses/BaseClasses/Ventilation/Vent_TTH_pressureNetwork_AirPerfectGass.mo [hunk 1]
```diff
@@ -0,0 +1,92 @@
+within IDEAS.Examples.TwinHouses.BaseClasses.Ventilation;
```
Comment (Mathadon) [inline IDEAS/Examples/TwinHouses/BaseClasses/Ventilation/Vent_TTH_pressureNetwork_AirPerfectGass.mo:1]: @kldjonge can you add documentation in the information section about why this class is needed? It sounds like you need a different Medium but you could have achieved that simply by redeclaring the Medium in the original model?

File: IDEAS/Examples/TwinHouses/BuildingO5_Exp1_1Port_PerfectGass.mo [hunk 2]
```diff
@@ -0,0 +1,31 @@
+within IDEAS.Examples.TwinHouses;
+model BuildingO5_Exp1_1Port_PerfectGass
+  "Model for simulation of experiment 1 for the O5 building"
```
Comment (Mathadon) [inline IDEAS/Examples/TwinHouses/BuildingO5_Exp1_1Port_PerfectGass.mo:3]: We already have `IDEAS.Examples.TwinHouses.BuildingO5_Exp1` for `Model for simulation of experiment 1 for the O5 building`. Can you adjust the documentation such that it is clear what the purpose of this model is? Please change both this comment and add a note in the documentation section.

File: IDEAS/Examples/TwinHouses/BuildingO5_Exp1_1Port_PerfectGass.mo [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      Tolerance=1e-06,
+      __Dymola_Algorithm="Cvode"),
+    __Dymola_experimentSetupOutput(events=false),
+    __Dymola_experimentFlags(
+      Advanced(GenerateVariableDependencies=false, OutputModelicaCode=false),
+      Evaluate=true,
+      OutputCPUtime=true,
+      OutputFlatModelica=false),
+    __Dymola_Commands(file=
+          "modelica://IDEAS/Resources/Scripts/Dymola/Examples/TwinHouses/BuildingO5_Exp1.mos"
```
Comment (Mathadon) [inline IDEAS/Examples/TwinHouses/BuildingO5_Exp1_1Port_PerfectGass.mo:29]: The .mos file indicated in this line its the unit test for the model BuildingOO5_Exp1. Can you add a similar mos file for this model and adjust the path accordingly? Then this model is unit tested together with the others. I can add the reference results.

File: IDEAS/Examples/TwinHouses/Interfaces/PartialTTHStructure_AirPerfectGass.mo [hunk 4]
```diff
@@ -0,0 +1,56 @@
+within IDEAS.Examples.TwinHouses.Interfaces;
+partial model PartialTTHStructure_AirPerfectGass
+  "Partial Model Based on Excel autogenerated"
+  extends IDEAS.Templates.Interfaces.BaseClasses.Structure(
+    redeclare each package Medium = Media.Specialized.Air.PerfectGas,
```
Comment (Mathadon) [inline IDEAS/Examples/TwinHouses/Interfaces/PartialTTHStructure_AirPerfectGass.mo:5]: can't you simply extend the other model and adjust the Medium?

**Review by Mathadon — CHANGES_REQUESTED** [review]
@kldjonge I have revised the code and I have a few questions. Can you have another look at these?

**Comment by kldjonge** [comment]
While trying to solve the comments, I found other issues with the original implementation of the Air.PerfectGass variation.
Also I can't make redeclaring the medium package top-level work. In the structure model it seems like not all sub-level packages are redeclared triggering errors. 

To be clear, we used the model without Air.PerfectGass for the paper.

**Comment by Mathadon** [comment]
@kldjonge do we need the PerfectGass variation in IDEAS if we did not use it for the paper? I see that you removed it, which is a valid solution for me!

**Comment by Mathadon** [comment]
@kldjonge I already merged these changes. Are there any others you'd like to add?

**Comment by kldjonge** [comment]
Should the CONTAM reference results be included for testing the airflows?

**Comment by Mathadon** [comment]
It would indeed be useful to have a Modelica model that implements an airflow network and that also contains a CombiTimeTable with matching CONTAM results for intercomparison! We could then include a unit test or even asserts that verify whether the error compared to contam is within some bounds. Could you add such a model?

**Comment by kldjonge** [comment]
Like this?

https://github.com/kldjonge/IDEAS/tree/1066_interzonal_KDJ

I noticed some deviations that aren't in the results generated in January though. E.g. In the W40.resDoor.m_flow, the dynamics are opposite but the steady state at the start is more or less correct, so changing the sign doesn't help in this case because then the whole graph is mirrored against the X-axis, and not around the 'center'

image

[CONTAMTwinHouseO5.txt](https://github.com/open-ideas/IDEAS/files/6610297/CONTAMTwinHouseO5.txt)

**Comment by Mathadon** [comment]
Yup, that's pretty much what I meant!

I'm getting different results than you though? 

image

The flow rates look like they're a factor two off. 

Possibly I made an error somewhere during refactoring :s

**Comment by Mathadon** [comment]
Should the orifice exponent for doors be 0.5 or 0.65?

And what is the mass flow rate convention for the CONTAM results? Is the flow rate positive when leaving or entering the building?

**Comment by kldjonge** [comment]
The exponent should be 0.5 and the discharge coefficient 0.78.
This can be changed but by default in CONTAM, positive flow is ambient to indoors.

**Comment by Mathadon** [comment]
Ok the convention in Modelica is often the other way around.

I plotted some comparisons of a few signals. Are these errors in line with the errors that you had before?
image
Possibly the errors changed because the CONTAM reference file uses outdated prescribed temperatures from an older IDEAS model?

**Comment by kldjonge** [comment]
That could explain the differences indeed. Modelica seems to peak higher at sudden events. But like you said, the indoor temperatures are not the same anymore where in the paper I made sure the CONTAM simulations was done after the Modelica simulation and temperatures where the same.

To compare with the paper, the flow rates plotted are:

- For external walls: W1,W7,W10
- For Internal walls: W37,W39, W42
- The open doors (resDoor): W40, W41, W42

**Comment by kldjonge** [comment]
The W10 results did fit much better though. (graph is is flipped here, units in m³/h)

image

**Pull request merged**